python-avd/pyavd/_eos_cli_config_gen/j2templates/eos/traffic-policies.j2

```diff
@@ -50,70 +59,64 @@ traffic-policies
 {%                     endif %}
 {%                 endfor %}
 {%                 if transient_values.counters | length > 0 %}
-      counter {{ transient_values.counters | join(' ') }}
+      counter {{ transient_values.counters | arista.avd.natural_sort | join(' ') }}
+      !
 {%                 endif %}
 {# MATCH SECTION #}
 {%                 for match in policy.matches %}
       match {{ match.name }} {{ match.type | lower }}
 {#                     -- SOURCE PREFIX MANAGEMENT -- #}
-{%                     if match.source is arista.avd.defined %}
-{%                         if match.source.prefixes is arista.avd.defined %}
-         source prefix {{ match.source.prefixes | join(' ') }}
-{%                         endif %}
-{%                         if match.source.prefix_lists is arista.avd.defined %}
-         source prefix field-set {{ match.source.prefix_lists | join(' ') }}
-{%                         endif %}
+{%                     if match.source.prefixes is arista.avd.defined %}
+         source prefix {{ match.source.prefixes | arista.avd.natural_sort | join(' ') }}
+{%                     elif match.source.prefix_lists is arista.avd.defined %}
+         source prefix field-set {{ match.source.prefix_lists | arista.avd.natural_sort | join(' ') }}
 {%                     endif %}
 {#                     -- DESTINATION PREFIX MANAGEMENT -- #}
-{%                     if match.destination is arista.avd.defined %}
-{%                         if match.destination.prefixes is arista.avd.defined %}
-         destination prefix {{ match.destination.prefixes | join(' ') }}
-{%                         endif %}
-{%                         if match.destination.prefix_lists is arista.avd.defined %}
-         destination prefix field-set {{ match.destination.prefix_lists | join(' ') }}
-{%                         endif %}
+{%                     if match.destination.prefixes is arista.avd.defined %}
+         destination prefix {{ match.destination.prefixes | arista.avd.natural_sort | join(' ') }}
+{%                     elif match.destination.prefix_lists is arista.avd.defined %}
+         destination prefix field-set {{ match.destination.prefix_lists | arista.avd.natural_sort | join(' ') }}
 {%                     endif %}
 {#                     -- PROTOCOL MANAGEMENT -- #}
 {%                     if match.protocols is arista.avd.defined %}
+{%                         set flags = {'bgp': False, 'neighbors': False, 'all': False} %}
 {%                         for protocol in match.protocols %}
-{#                             -- destination port-list -- #}
-{%                             if protocol.dst_port is arista.avd.defined and protocol.protocol | lower in ['tcp','udp'] %}
+{%                             if protocol.protocol is arista.avd.defined and protocol.protocol not in ['neighbors', 'bgp'] and not flags.bgp and not flags.neighbors %}
+{%                                 set _ = flags.update({'tcp': True}) %}
```

Comment:
`flags.tcp` this is not being used anywhere

Reply:
you might want to set `flags.all: true` here ?

Reply:
updated